docs/spec/v1beta1/provider.md

```diff
@@ -730,3 +732,141 @@ kubectl create secret generic webhook-url \
+
+* The `client_payload` contains the event data from the notification controller and looks like this:
+
+```yaml
+{
+  involvedObject: {
+    apiVersion: kustomize.toolkit.fluxcd.io/v1beta2,
+    kind: Kustomization,
+    name: podinfo,
+    namespace: flux-system,
+    resourceVersion: 426573,
+    uid: b9b8554d-be26-4a3d-a97f-65f3276a097a
+  },
+  message: Deployment/podinfo/podinfo configured,
+  metadata: {
+    revision: main/96139968ca46b53462d1bf94de410a811d2026a1,
+    summary: "staging (us-west-2)"
+  },
+  reason: Progressing,
+  reportingController: kustomize-controller,
+  reportingInstance: kustomize-controller-79464d8dc5-nb9c4,
+  severity: info,
+  timestamp: 2022-04-20T12:20:28Z
+}
+```
+
+#### Setting up a GitHub workflow
+
+You need to add `repository_dispatch` trigger to trigger your GitHub workflow using the Flux notification with the relevant event type:
+
+```yaml
+name: test-github-dispatch-provider
+on:
+  repository_dispatch:
+    types: [Kustomization/podinfo.flux-system]
+```
+
+Assuming that we deploy all Flux kustomization resources in the same namespace, it will be useful to have a unique kustomization resource name for each application. This will allow you to use only `event_type` to trigger tests for the exact application.
+
+Let's say we have following folder structure for applications kustomization manifests:
+
+```bash
+apps/
+├── app1
+│   └── overlays
+│       ├── production
+│       └── staging
+└── app2
+    └── overlays
+        ├── production
+        └── staging
+```
+
+You can then create a flux kustomization resource for the app to have unique `event_type` per app. The kustomization manifest for app1/staging:
+
+```yaml
+apiVersion: kustomize.toolkit.fluxcd.io/v1beta2
+kind: Kustomization
+metadata:
+  name: app1
+  namespace: flux-system
+  spec:
+    path: "./app1/staging"
+```
+
+You would also like to know from the notification which cluster is being used for deployment. You can add the `spec.summary` field to the Flux alert configuration to mention the relevant cluster:
+
+```yaml
+apiVersion: notification.toolkit.fluxcd.io/v1beta1
+kind: Alert
+metadata:
+  name: github-dispatch
+  namespace: flux-system
+spec:
+  summary: "staging (us-west-2)"
+  providerRef:
+    name: github-dispatch
+  eventSeverity: info
+  eventSources:
+    - kind: Kustomization
+      name: 'podinfo'
+```
+
+Now you can the trigger tests in the GitHub workflow for app1 in a staging cluster when the app1 resources defined in `./app1/staging/` are configured by Flux:
+
+```yaml
+name: test-github-dispatch-provider
+on:
+  repository_dispatch:
+    types: [Kustomization/podinfo.flux-system]
+jobs:
+  run-tests-staging:
+    if: github.event.client_payload.metadata.summary == 'staging (us-west-2)'
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Run tests
+      run: echo "running tests.."
+```
+
+### Setting up the GitHub dispatch provider
```

Comment:
Let's move this before "Setting up a GitHub workflow" please.

Reply:
Fixed